src/main/java/sqltoregex/visitor/SelectVisitorJoinToWhere.java

```diff
@@ -23,10 +24,13 @@ public void visit(PlainSelect plainSelect) {
         for (Join join : joinList) {
             //In MySQL, it is not possible to have multiple ON, all are concatenated using "AND" that's why the
             // OnExpressionList has always 1 Element for our useCase
-            if (!join.getOnExpressions().stream().findFirst().isPresent()) {
+            if (join.getOnExpressions().stream().findFirst().isEmpty()) {
                 continue;
             }
             if (plainSelect.getWhere() == null) {
+                if(join.getOnExpressions().stream().findFirst().isEmpty()){
+                    throw new NoSuchElementException();
```

Comment:
I think this is wrong.:

Why there is an Join without conditions. (Neither in ON nor in WHERE)
Joinlist != null
getWhere == null
getOnExpressions() is empty

This will throw an Exception with this addition and abort/stop converting with an error message to the user, although its in my eyes a valid input.
(Since the plan was to use the SelectVisitorJoinToWhere in any case and compare the too parsings)